### ISSUE TITLE
TSLint complains about `> {` with generics and one-line.

### ISSUE THREAD

**Issue by fongandrew** [palantir/tslint#3084]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.4.1
- __Running TSLint via__: (pick one) VSCode

#### TypeScript code being linted

```ts
// code snippet
export interface MyInterface<
  R extends string,
  D extends string,
  O extends string
> {
   /* ... */
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "one-line": [true,
      "check-open-brace"
    ]
  }
}
```

#### Actual behavior
TSLint complains about `misplaced opening brace (one-line)`  on the `> {` line

#### Expected behavior
No complaint. This is essentially the same as the `> (` in the following code, which doesn't trigger a TSLint error.

```ts

function test<
  R extends string,
  D extends string,
  O extends string
> (
  required: R,
  default: D,
  optional: O
) {
  /* ... */
}
```

**Comment by ajafff** [comment]
Duplicate of #2952 and will be fixed by #3004
